RFC: usb: device: Allow limited amount of tries for write

finikorg opened the pull request:
Set limited amount of tries to write to USB control endpoint EP0.

dcpleung committed:
soc: intel_s1000: add _heap_sentry to linker script

Newlib requires a _heap_sentry so we add it to the linker script,
similar to what other xtensa linker scripts.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
cmake: xtools: xtensa: use SoC name in path to toolchain

Xtensa requires building a new toolchain for a specific SoC.
By default xtools built Xtensa toolchains all have prefix of
xtensa-zephyr-elf. In order to distinguish different toolchains,
they are now placed in their own directories under their SoC
name. This allows us to have multiple Xtensa toolchains
targeting multiple SoCs.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
soc: intel_s1000: define default MEMCTL reg value

When not using XCC, XCHAL_CACHE_MEMCTL_DEFAULT is not defined
which results in some variables now being able to be defined.
So define them.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
boards: intel_s1000_crb: allows to build with xtools also

This allows applications targeting intel_s1000_crb to be
compiled with custom toolchain.

Also copy the testing ignore_tags from qemu_xtensa for
net and bluetooth.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
samples: intel_s1000/i2s: fix uninitialized use warning

One of the return variable is declared and used, but never
assigned values. So fix it.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
wip

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
arch: xtensa: use asm for _xt_set_intset/_xt_set_intclear

Use assembly for _xt_set_intset() and _xt_set_intclear() instead of
calling into the Xtensa HAL, allowing these to be inlined.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

carlescufi reviewed (commented):
Can you please add a more precise explanation in the commit message?

masz-nordic commented:
Number of retries could be a `Kconfig` value.

finikorg commented:
> Number of retries could be a `Kconfig` value.

This is kind of workaround, I think it should not be in Kconfig

nashif committed:
manifest: add sofproject repo as a module

This is the first step, we need to add zephyr/module.yml into the zephyr
branch of the sofproject to activate it and point it to cmake/kconfig to
enable the build.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Vudentz committed:
Modules: Add Kconfig for SOF

This adds the initial Kconfig for SOF.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Commit:
soc: xtensa: Create SoC definition for Audio DSP on Intel Apollo Lake

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
boards: xtensa: Create board definition for Audio DSP on Up Squared

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Vudentz committed:
subsys: Add initial support for using SOF

This introduces support for Sound Open Firmware as audio subsystem.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Commit:
updated hello_world

Vudentz committed:
Audio: Add sample for testing a static pipeline

This samples enables CONFIG_SOF which currently initialize a static
pipeline.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Commit:
disable SoC init

Commit:
enable CONFIG_TRACE

dcpleung committed:
intel_apl_adsp: change memory mapping to match SOF firmware

This changes the memory mapping to match SOF firmware, such that
the SOF bootloader can jump directly to Zephyr's start code.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
log: Add Intel ADSP logging backend

Add logging backend for Intel ADSP.

finikorg committed:
soc/intel_apl_adsp: Fix build errors

Fix build error which arises from assigning string to uint16_t.

finikorg committed:
log: log_backend_adsp: Use ringbuffer

Use ringbuffer for logging to memory

dcpleung committed:
intel_apl_adsp: enable Xtensa simulator logging backend

This allows using the logging subsystem when running under QEMU.
Note that "-semihosting" needs to be passed to QEMU command line
for the "simcall" instruction to work. Or else it would simply
be an illegal instruction.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
log: log_backend_adsp: Add magic and log_id to log slot

Add magic number which is needed to be able to check the presence of
message and log message id which is needed for log sorting.

dcpleung committed:
intel_apl_adsp: enable reset vector even with SOF bootloader

This enables using Zephyr's Xtensa reset vector even when being
bootstrapped with SOF boot loader. This allows us to setup
the correct environment instead of inheriting one set by
the SOF bootloader.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
hello_world: remove old logging

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
samples/hello_world: Add logging to the sample

Add logging to sample

dcpleung committed:
hello_world: enable ADSP logging backend

Also disable the Xtensa SIM logging backend as this won't
work with actual hardware.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: enable Xtensa simulator logging backend

This allows using the logging subsystem when running under QEMU.
Note that "-semihosting" needs to be passed to QEMU command line
for the "simcall" instruction to work. Or else it would simply
be an illegal instruction.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

Vudentz committed:
audio: sof: Enable CONFIG_LOG_BACKEND_XTENSA_SIM

This enables CONFIG_LOG_BACKEND_XTENSA_SIM on the SOF so it is able to
output logs when running under qemu.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

dcpleung committed:
DNM: intel_apl_adsp: signal host firmware is ready

Hack around FW ready signal.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
DNM: intel_apl_adsp: fake ABI version for driver to load

Fake the ABI version so the driver will finish loading
the firmware onto DSP, instead of crashing.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: setup host windows

This sets up the DMA Window Registers to expose certain memory
area for the host.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: setup clocks and power management at soc_init()

This sets up the clocks for both memory domains and disabling
power gating for PLL and process cores.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
up_squared_adsp: adds a simply tool to read log lines...

...exposed in the debugfs nodes on the Linux host side.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
up_squared_adsp: tools: Fix compilation warnings

Trivial compilation warning fixes

finikorg committed:
docs: Update README.md to SOF audio sample

Update repository and remove old methods.

finikorg committed:
docs: Add README.md to SOF audio sample

Add README.md

finikorg committed:
up_squared_adsp: tools: Add python logtool

Add simple python logtool to extract logs from memory ring buffer, at
the moment hardcoded.

dcpleung committed:
subsys: sof: move SoC init code into here

Moves the SoC SOF initialization code from intel_apl_adsp into
the SOF subsystem. Those code is mainly responsible for
notifying the host SOF driver whether the firmware is up and
running. So it is logical for init to be done inside the SOF
subsystem.

This also moves the platform specific header files from
the SOF module into the SoC directory, as the arch/SoC/board
definitions should be defined inside Zephyr tree. Due to
the header shuffling, the ADSP logging backend now requires
CONFIG_SOF as the mailbox #defines are inside the SOF
module code.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
samples: sof: Move logging backend to overlay

Moving logging backend to overlay allows to select right one when
building with west with -DOVERLAY_CONFIG parameter.

finikorg committed:
log_backend_adsp:  Fix compiler warning

Make type cast to avoid warnings

dcpleung committed:
samples: hello_world: revert to original hello_world

Should use samples/audio/sof instead of hello_world. So revert
hello_world back to original version.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys: audio/sof: initialize clock in SOF

During SOF initialization, the clock structures also need to
be initialized in the SOF module. So call the correct
function to do that.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: clean up interrupt en-/dis-abling code in SoC

The code originally comes from intel_s1000 which has a 3rd level
interrupt controller. However, intel_apl_adsp only has 2 levels.
So clean up the code.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: clean up intel_s1000 remnant in SoC code

The code was originally copied from intel_s1000 which contains
code to access hardware which does not exist on intel_apl_adsp.
So remove those code.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: make SOC_AGGREGATE_IRQ macro straight forward

Zephyr's multiple level interrupt encodings requires 2nd level
and up be off by 1 in the encodings. The SOC_AGGREGATE_IRQ macro
requires developers to increment the interupt number before
passing the number to macro. This is error prone so do
the increment in the macro itself, and developers can simply
pass in the actual interrupt number.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
samples: audio/sof: enable LOG_PRINTK

This allows printk() to be printed to the ADSP logging
backend, and thus readable from Linux host.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: update interrupt handler header file

The file was from intel_s1000. As Xtensa cores have different
configurations, we need to generate a new one for intel_apl_adsp.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: enable multi-level interrupts if SOF

Enable multi-level interrupts and 2nd level interrupts
if CONFIG_SOF=y, since SOF requires multi-level interrupts.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
boards: up_squared_adsp: disable CONFIG_3RD_LEVEL_INTERRUPTS

There is no interrupt controller behind the CAVS interrupt controller
so disable the 3rd level interrupt kconfig.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
xtensa: partially enable dynamic interrupts

Without multi-level interrupts, enabling dynamic interrupts
on Xtensa when CONFIG_XTENSA_ASM2=y is trivial. However,
with multi-level interrupts, the vectors in the ISR vector
table are not placed linearly. So when multi-level
interrupts is enabled, defer to SoC for dynamic interrupt
support, the same irq_en-/dis-able().

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: enable dynamic interrupts if SOF

This implements the necessary functions to enable dynamic
interrupts with multi-level interrupts.

Also sets the kconfig to enable dynamic interrupts if
CONFIG_SOF is also enabled, since the SOF module
requires installing ISR at runtime.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
samples/audio: Update README

Add mention how to build audio/sof sample for Qemu and up_squared
board.

finikorg committed:
sof: Add dai_init and placeholder

Add dai_init() and stub for now.

finikorg committed:
sof: Add dai_init and placeholder

Add dai_init() and stub for now.

finikorg committed:
log: Decrease buf size and increase number of logs

This decreases buffer for logs to 60 bytes and increases number of
logs to 128.

finikorg committed:
xtensa: tools: Update logging tools for the new buffer

Update logging tools to have smaller buffer and more log lines.

finikorg committed:
log: adsp: Fix logging for long strings

Fix long string logging and fix cache writeback.

jfischer-phytec-iot commented:
This would probably fix one of the bugs I found in request handling, I will open a issue.

jfischer-phytec-iot commented on subsys/usb/usb_device.c:
```diff
@@ -1160,15 +1155,14 @@ int usb_disable(void)
+		ret = usb_dc_ep_write(ep, data, data_len, bytes_ret);
+	} while (ret == -EAGAIN && tries--);
 
-		k_yield();
```
I prefer to keep k_yield or k_sleep here.

jfischer-phytec-iot commented on subsys/usb/usb_device.c:
```diff
@@ -1160,15 +1155,14 @@ int usb_disable(void)
 {
-	while (true) {
-		int ret = usb_dc_ep_write(ep, data, data_len, bytes_ret);
+	int tries = 10;
```
Number of tries should have Kconfig option, maybe range 3 10?

finikorg committed:
xtensa: adsp: Remove DMIC configuration

Remove DMIC configuration from sof build.

finikorg committed:
sof: Add DMAC initialization

Add DMAC initialization to sof_init()

dcpleung committed:
intel_apl_adsp: implement z_soc_irq_is_enabled()

This provides an implemention for z_soc_irq_is_enabled()
as it is needed for multi-level interrupts.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
irq_nextlevel: add API to query if an IRQ line is enabled

There is an API to query if any IRQ is enabled but there is
none to query individual IRQ line. So add one.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_s1000: implement z_soc_irq_is_enabled()

This provides an implemention for z_soc_irq_is_enabled()
as it is needed for multi-level interrupts.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: start DSP wall clock at init

Simply call platform_timer_start() to start the DSP wall
clock during SOF initialization.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: initialize interrupts at SOF init

The interrupt masks need to be set correctly at time of
SOF initialization.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: remove DAI and DMA stubs

The DAI and DMA drivers in SOF are being enabled,
so remove the stubs.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: remove setting magic number at init

The magic number was used at the very beginning of board
bring-up to signal the firmware was running. There is no
need for this anymore as the initialization is at a point
where a connected HiFiBerry DAC has its green indicator
light lit. So remove the magic number.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: initialize schedulers earlier

The official SOF initializes the schedulers earlier so move
the function call ealier here.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
kernel: allow delayed work with delays in microseconds

This allows delayed work with delays in microseconds, rather
than simply milliseconds.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
intel_apl_adsp: increase ticks per second

When scheduling works, SOF requires delays in microsecond range.
So increase the number of ticks per second to 1 tick per
10 microseconds (instead of 1 tick per 1us to avoid too many
interrupts per second).

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
samples/audio: sof: enlarge heap memory pool size

SOF needs lots of memory to buffer audio signals, so
make the heap memory pool larger.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
modules: sof: add SOF_USE_DMIC kconfig

Not all platforms support DMIC, so add a kconfig to selectively
include codeto compile.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
Revert "intel_apl_adsp: increase ticks per second"

This reverts commit 1f03148fc7cdd3a9976beb6682de0410a10f43ae.

Does not seem to affect playback. So revert it for now
as this has potential to cause too many scheduler events.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
subsys/audio: sof: use mem_slab for buffers

Zephyr's k_mem_pool does not allocate buffers that are cache line
and/or DMA transfers aligned. The manual cache line manipulations
has the potential to disrupt the k_mem_pool's own tracking
information, and thus causing exceptions in freeing the memory.
So the memory allocator is modified to use memory slab for
buffers, and it needs to be initialized.

Shrink the size of k_mem_pool as buffers are not being allocated
from there.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

finikorg committed:
up_squared_adsp: Add custom target for ADSP

Adding special section is required by image creation tool.

finikorg committed:
xtensa: tools: Update python logger tool

Allow to take arguments for etrace file.

finikorg committed:
RFC: intel_apl_adsp: Update headers

Move code from cavs/* to the headers.

finikorg committed:
up_squared_adsp: tools: Update logtool

Add log sorting in logger tool.

ioannisg commented:
Moving to next release since the review was not finalized for v.2.0.0

finikorg committed:
log: Fix CMakeLists file

Fix brace

dbkinder commented on include/kernel.h:
```diff
@@ -2918,7 +2920,42 @@ extern void k_delayed_work_init(struct k_delayed_work *work,
+ *
+ * @param work_q Address of workqueue.
+ * @param work Address of delayed work item.
+ * @param delay Delay before submitting the work item (in microseconds).
```
actual param name is ``us_delay`` in the code below

dbkinder commented on boards/xtensa/up_squared_adsp/doc/index.rst:
```diff
@@ -0,0 +1,4 @@
+.. _Up_Squared_Audio_DSP:
+
+Up Squared Audio DSP
+####################
```
need some doc content here , but you know that :)
